tests/internal/test_wrapping.py

```diff
@@ -926,3 +927,39 @@ def foo():
 
     new_method_count = len([_ for _ in gc.get_objects() if type(_).__name__ == "method"])
     assert new_method_count <= method_count + 1
+
+
+def test_wrap_once():
+    c = 0
+
+    def wrapper(f, args, kwargs):
+        nonlocal c
+        c += 1
+        return f(*args, **kwargs)
+
+    def foo():
+        pass
+
+    wrap_once(foo, wrapper)
+
+    for _ in range(10):
+        foo()
+
+    assert c == 1
+
+
+def test_wrapping_context_lazy():
+    free = 42
+
+    def foo():
+        return free
+
+    (wc := DummyWrappingContext(foo)).wrap_lazy()
+
+    assert not DummyWrappingContext.is_wrapped(foo)
```

Comment:
This is great, but how do we uninstrument when the lazy decorator has been installed but hasn't been executed?